Merger/src/main/kotlin/lushu/Merger/Lattice/NodeFactory.kt

```diff
@@ -129,7 +129,7 @@ class NodeFactory(
     ): PwsetNode {
         val nodeID = globalNodeID
         globalNodeID++
-        val newNode = PwsetNode(nodeID, blacklist, charset, interval)
+        val newNode = PwsetNode(nodeID, blacklist, charset, interval, sensitive)
```

Comment:
This is where the bug was

Reply:
Hi Alex. After reviewing the code changes, I believe that the error you mentioned could be one of the problems. However, there are a few others that also need to be addressed.

Firstly, when using the function mi.getNodeFactory().buildIntervalNodes(string, flagSensitive), it returns a list<IntervalNodes> where each IntervalNode is respective to one of the characters in the string. For example, if the string is "test", it would return [t][e][s][t], where each character is an IntervalNode. In my opinion, the best solution would be to return a single IntervalNode within the list, [test].

Secondly, when merging two strings with merge.merge(listToken1,listToken2), it returns a listToken that combines the tokens where possible, but it doesn't consider sensitivity. This means that if listToken2 is a sensitive token and listToken1 is a non-sensitive token, they will still be combined, resulting in a list with a non-sensitive token. In this case, I would expect the function to return a list with two tokens, one respective to the sensitive token and the other to the non-sensitive token.